orchestra/workflow/load.py

```diff
@@ -0,0 +1,161 @@
+from django.db import transaction
+
+from orchestra.models import Certification
+from orchestra.models import Workflow
+from orchestra.models import WorkflowStep
+from orchestra.models import WorkflowVersion
+from orchestra.workflow.directory import parse_workflow_directory
+from orchestra.workflow.exceptions import WorkflowError
+
+
+@transaction.atomic
+def load_workflow(workflow_directory, versions=None, force=False):
+    data = parse_workflow_directory(workflow_directory)
+
+    # Create the workflow object if it doesn't exist
+    workflow_data = data['workflow']
+    workflow, workflow_created = create_or_update(
+        Workflow,
+        {'slug': workflow_data['slug']},
+        {
+            'name': workflow_data['name'],
+            'description': workflow_data['description'],
+            'code_directory': workflow_directory
+        }
+    )
+
+    # Create all certifications for the workflow
+    for certification_data in workflow_data['certifications']:
+        create_or_update(
+            Certification,
+            {
+                'slug': certification_data['slug'],
+                'workflow': workflow
+            },
+            {
+                'name': certification_data['name'],
+                'description': certification_data['description'],
+            }
+        )
+
+    # Create the certification dependencies once all certs are in the db
+    for certification_data in workflow_data['certifications']:
+        cert = Certification.objects.get(
+            slug=certification_data['slug'],
+            workflow=workflow
+        )
+        required_cert_slugs = certification_data['required_certifications']
+        required_certs = Certification.objects.filter(
+            workflow=workflow,
+            slug__in=required_cert_slugs
+        )
+        if required_certs.count() != len(required_cert_slugs):
+            raise WorkflowError(
+                'Certification {} requires non-existent certification.'
+                .format(certification_data['slug']))
+        cert.required_certifications = list(required_certs)
```

Comment:
Yeah, but this is about dependencies. As in, I have Cert C, which has no required_certs. But someone reloads the workflow (or loads a new version), and the manifest now says that Cert C requires Cert B, which I don't have. Now I have an 'illegal' certification.

Should we: 
a) allow this situation, because it seems rare (and we want to let dependencies evolve over time).
b) throw an exception and refuse to change the certification dependencies.